custom_components/zwave_mqtt/binary_sensor.py

```diff
@@ -21,26 +22,261 @@
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
 
 from .const import DATA_UNSUBSCRIBE, DOMAIN
-from .entity import ZWaveDeviceEntity, create_device_name
+from .entity import ZWaveDeviceEntity
 
 _LOGGER = logging.getLogger(__name__)
 
-DEVICE_CLASS_MAPPING = {
-    # Mapping from Value Index in Notification CC to device class
-    1: DEVICE_CLASS_SMOKE,
-    2: DEVICE_CLASS_GAS,
-    3: DEVICE_CLASS_GAS,
-    4: DEVICE_CLASS_HEAT,
-    5: DEVICE_CLASS_MOISTURE,
-    6: DEVICE_CLASS_SAFETY,
-    7: DEVICE_CLASS_SAFETY,
-    8: DEVICE_CLASS_POWER,
-    9: DEVICE_CLASS_PROBLEM,
-    10: DEVICE_CLASS_PROBLEM,
-    14: DEVICE_CLASS_SOUND,
-    15: DEVICE_CLASS_MOISTURE,
-    18: DEVICE_CLASS_GAS,
-}
+# Value Indexes for Notification CommandClass
+# TODO: Move to enum in the library
+NOTIFICATION_SMOKE_ALARM = 1
+NOTIFICATION_CARBON_MONOOXIDE = 2
+NOTIFICATION_CARBON_DIOXIDE = 3
+NOTIFICATION_HEAT = 4
+NOTIFICATION_WATER = 5
+NOTIFICATION_ACCESS_CONTROL = 6
+NOTIFICATION_HOME_SECURITY = 7
+NOTIFICATION_POWER_MANAGEMENT = 8
+NOTIFICATION_SYSTEM = 9
+NOTIFICATION_EMERGENCY = 10
+NOTIFICATION_CLOCK = 11
+NOTIFICATION_APPLIANCE = 12
+NOTIFICATION_HOME_HEALTH = 13
+NOTIFICATION_SIREN = 14
+NOTIFICATION_WATER_VALVE = 15
+NOTIFICATION_WEATHER = 16
+NOTIFICATION_IRRIGATION = 17
+NOTIFICATION_GAS = 18
+
+NOTIFICATION_TYPE = "index"
+NOTIFICATION_VALUES = "values"
+NOTIFICATION_DEVICE_CLASS = "device_class"
+NOTIFICATION_SENSOR_ENABLED = "enabled"
+
+
+# Translation from values in Notification CC to binary sensors
+# https://github.com/OpenZWave/open-zwave/blob/master/config/NotificationCCTypes.xml
+NOTIFICATION_SENSORS = [
+    {
+        # Index 1: Smoke Alarm - Value Id's 1 and 2
+        NOTIFICATION_TYPE: NOTIFICATION_SMOKE_ALARM,
+        NOTIFICATION_VALUES: [1, 2],
```

Comment:
So does this result in two different binary sensors?

1. `binary_sensor.<device>_smoke_alarm_smoke_detected_at_location`
2. `binary_sensor.<device>_smoke_alarm_smoke_detected_at_unknown_location`

But then I noticed you are stripping from `at` or `on` and after, which would cause both of these to have the same name. Are you assuming a device would not support both values, or what would happen if it did?

For these values in particular, I think it would make sense for them to always be the same sensor. The location event data should be an attribute?

Reply:
As far as I understand the specs (and seen it on my own devices) you will not have both sensors at the same time present. Also I have never seen the location filled.
Do you say otherwise ?

Reply:
Nope, I can't say either way. I don't see anything that says they are mutually exclusive though. I only have an old smoke alarm that is V1, so it's much worse.

Reply:
OK, I'll put in some comments in the code. If someone actually owns a device that has both values present he will get 2 binary sensors with the same name and report that. In that case we can extend the code. I have a strong feeling that is never going to happen :-)

Reply:
Added some comments in the code.